capstone/src/components/User/Adventurer/ExperienceInfo/ExperienceInfo.jsx

```diff
@@ -25,6 +51,42 @@ export default function ExperienceInfo({ restaurants }) {
     ? getPhotoReference(currentRestaurant)
     : null;
 
+  // get active feedback information
+  useEffect(() => {
+    if (currentRestaurant) {
+      getActiveFeedback(currentRestaurant.place_id).then((res) =>
+        setActiveFeedback(res.data),
+      );
+      getIsRated(username, currentRestaurant.place_id).then((res) =>
+        setIsRated(res.data),
+      );
+    }
+  }, [setActiveFeedback, currentRestaurant, username, setIsRated]);
+
+  function resetRatingForm() {
+    return activeFeedback.map((feedback) => ({
+      feedbackId: feedback.objectId,
+      score: 0,
+      comment: '',
+    }));
+  }
+
+  // update State for the newReview, if is not rated
+  useEffect(() => {
+    if (!isRated && activeFeedback) {
```

Comment:
If activeFeedback is set before isRated, this block will always be called. ex.

1. initial state: isRated = false, and activeFeedback = []
2. getActiveFeedback() returns. activeFeedback = [{...}]
3. React component state updates. setNewReview is called.
4. getIsRated() returns. isRated = true.
5. whoops! we already called setNewReview before

Reply:
wow! I never saw that coming! 
I decided to take out activeFeedback, that way only when isRated changes it is triggered